react/features/virtual-background/components/VirtualBackgroundDialog.js

```diff
@@ -154,36 +143,42 @@ function VirtualBackground({ _selectedThumbnail, dispatch, t }: Props) {
             const url = await resizeImage(reader.result);
             const uuId = uuid.v4();
 
-            isloading(true);
             setStoredImages([
                 ...storedImages,
                 {
                     id: uuId,
                     src: url
                 }
             ]);
-            await dispatch(
-                toggleBackgroundEffect({
-                    backgroundType: 'image',
-                    enabled: true,
-                    url,
-                    selectedThumbnail: uuId
-                })
-            );
-            isloading(false);
+            setOptions({
+                backgroundType: 'image',
+                enabled: true,
+                url,
+                selectedThumbnail: uuId
+            });
         };
         reader.onerror = () => {
             isloading(false);
             logger.error('Failed to upload virtual image!');
         };
     };
 
+    const applyVirtualBackground = async () => {
+        isloading(true);
+        await dispatch(toggleBackgroundEffect(options, _jitsiTrack));
+        await isloading(false);
+        dispatch(hideDialog());
+    };
+
     return (
         <Dialog
             hideCancelButton = { true }
```

Comment:
We should now enable cancelling the dialog, shouldn't we? In case you don't want to make any changes.